### ISSUE TITLE
Loaded model is different from saved model

### ISSUE THREAD

**Issue by Skeltch** [Microsoft/CNTK#1596]
After training a model I test the model and print the metric returned from epoch_summary. However when I save the model and reload it the value returned is different, and across the different trainings, the returned values are very similar. I am unsure why something like this would be happening, however these are the results from 4 different instances of training and testing versus loading and testing. Any help would be appreciated.

Results from running

1.

> Finished Epoch[16 of 300]: [Training] loss = 2.290549 * 199931, metric = 229.1% * 199931 109.009s (1834.1 samples per second);
> Training Done!
> Finished Epoch[1 of 0]: [Evaluation] loss = 0.000000 * 105008, metric = 315.9% * 105008 14.507s (7238.5 samples per second);
> 3.1588833091209234

> Load
> 6.72678667609328

2.

> Finished Epoch[1 of 0]: [Evaluation] loss = 0.000000 * 105008, metric = 277.5% * 105008 9.661s (10868.9 samples per second);
> 2.775055581179585

> Load
> 6.726787232923995

3.

> Finished Epoch[1 of 0]: [Evaluation] loss = 0.000000 * 105008, metric = 315.9% * 105008 14.550s (7217.2 samples per second);
> 3.1588833091209234

> Load
> 6.726787175962189

4.

> Finished Epoch[32 of 300]: [Training] loss = 1.610496 * 199990, metric = 161.0% * 199990 85.438s (2340.8 samples per second);
> Training Done!
> Finished Epoch[1 of 0]: [Evaluation] loss = 0.000000 * 52504, metric = 263.0% * 52504 5.472s (9595.0 samples per second);
> 2.6300985001217168

> Load
> Finished Epoch[1 of 0]: [Evaluation] loss = 0.000000 * 52504, metric = 680.4% * 52504 5.428s (9672.8 samples per second);
> 6.803679589906989

Here's the code for testing and evaluating the model

	def evaluate(reader, model):
		#input
		features = cntk.blocks.Input(input_dim, is_sparse=False)
		labels = cntk.blocks.Input(label_dim, is_sparse=False)
		
		#output
		netoutput = model(features)

		#loss and metrics
		ce = cntk.ops.squared_error(netoutput, labels)
		pe = cntk.ops.squared_error(netoutput, labels)

		# process minibatches and perform evaluation
		lr_schedule = cntk.learner.learning_rate_schedule(1, cntk.learner.UnitType.minibatch)
		momentum_as_time_constant = cntk.learner.momentum_as_time_constant_schedule(0)
		dummy_learner = cntk.learner.adam_sgd(netoutput.parameters, 
								 lr=lr_schedule, momentum=momentum_as_time_constant, low_memory=True)
		evaluator = cntk.Trainer(netoutput, (ce, pe), [dummy_learner])
		progress_printer = cntk.ProgressPrinter(tag='Evaluation', num_epochs=0)

		input_map = {features: reader.streams.features, labels: reader.streams.labels}

		while True:
			minibatch_size = 500
			data = reader.next_minibatch(minibatch_size, input_map=input_map)
			if not data:                                 # until we hit the end
				break
			metric = evaluator.test_minibatch(data)
			progress_printer.update(0, data[labels].num_samples, metric) # log progress
		loss, metric, actual_samples = progress_printer.epoch_summary(with_metric=True)
		print (metric)

		return loss, metric

To load and save the model we create the model

	def create_model():
		with cntk.layers.default_options(initial_state=0.1):
			return cntk.models.Sequential([
				cntk.layers.BatchNormalization(),
				cntk.layers.Recurrence(cntk.blocks.LSTM(lstm_dim), go_backwards=False),
				cntk.layers.Recurrence(cntk.blocks.LSTM(lstm_dim2), go_backwards=False),
				cntk.layers.Dense(label_dim)
			])
and then use
`model.save("..")`
and
`model.load("..")`

**Comment by Master76** [comment]
Hi @Skeltch ,

Open your saved model using a notepad, does it totally look like a binary file, or you can figure out some words and expressions in it?
if it looks like behind, you probably did not trained the model successfully.
```plain text

version("
uidBCompositeFunction1710��
primitive_functionsҥ	Zʥ
�
b�
version(
uidB
SumAll1706
typeBPrimitiveFunction
```
If so, upgrade to cntk 2.0 beta15, and try again.

**Comment by Skeltch** [comment]
Opening it as text file, the first few lines resemble what you showed. I will update CNTK and try again. I'm curious, is it a CNTK bug, or is my model incorrectly formulated?

Thanks!

**Comment by Master76** [comment]
Hi @Skeltch ,

This issue had made my puzzled for a quite long time. I was using Recurrence & LSTM too. I'm sure it is a bug, because the CNTK python Layers library was refactoring that time. They probably released it without test by mistake. 

I updated to beta 15, and it now goes well. My codes:
```python
my_model = Sequential ([
                Dense(hidden_dim, activation=C.sigmoid),
                Recurrence(LSTM(project_dim, cell_dim, enable_self_stabilization=True)),
                Recurrence(LSTM(project_dim, cell_dim, enable_self_stabilization=True)),
                Dense(outputs)])
z = my_model(features)
```
It seems that formerly the training course just do nothing, and the model saved is just the pre-trained model.

**Comment by Skeltch** [comment]
Yup that seemed to do the trick. Thanks for the help!
